### PR TITLE
Av/lirc driver

### PR DESCRIPTION
**Pull request by eljuguetero**
prematurely closed, reopened

### CODE REVIEW

File: src/driver_lirc.cpp [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+namespace matrix_malos {
+
+const bool kLircDriverDebugEnabled = true;
+
+bool LircDriver::ProcessConfig(const DriverConfig& config) {
+  LircParams lirc(config.lirc());
+
+  if (lirc.device() == "" || lirc.command() == "" ||
+      !isValidLircSymbol(lirc.device()) || !isValidLircSymbol(lirc.command())) {
+    std::string error_msg("device or command parameter is missing or invalid");
```
Comment (arhuaco) [inline src/driver_lirc.cpp]: You could use a single line for this.
zmq_push_error_->Send(kLircDriverName + ": device or command parameter is missing or invalid.")

File: src/driver_lirc.cpp [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    zmq_push_error_->Send(error_msg);
+    return false;
+  }
+
+  if (kLircDriverDebugEnabled) {
+    std::cout << "device :" << lirc.device() << "\t";
+    std::cout << "command:" << lirc.command() << std::endl;
+  }
+
+  std::string str_irsend;
```
Comment (arhuaco) [inline src/driver_lirc.cpp]: You can declare and initialize in the same line.

File: src/driver_lirc.cpp [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  str_irsend = "irsend SEND_ONCE " + lirc.device() + " " + lirc.command();
+
+  if (system(str_irsend.c_str()) == -1) {
+    std::string error_msg("ir command failed");
```
Comment (arhuaco) [inline src/driver_lirc.cpp]: You can replace 3 lines with one.

File: src/driver_lirc.cpp [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    error_msg += kLircDriverName;
+    zmq_push_error_->Send(error_msg);
+    return false;
+  }
+  return true;
+}
+
+bool LircDriver::isValidLircSymbol(const std::string word) {
```
Comment (arhuaco) [inline src/driver_lirc.cpp]: const std::string&

File: src/js_test/test_ir_remote.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+var configSocket = zmq.socket('push')
+configSocket.connect('tcp://' + creator_ip + ':' + creator_everloop_base_port /* config */)
+
+function sendIrCommand() {
+  var ir_cfg_cmd = new matrixMalosBuilder.LircParams
+  ir_cfg_cmd.set_device('SONY')
+  ir_cfg_cmd.set_command('KEY_POWER')
+
+  var config = new matrixMalosBuilder.DriverConfig
+  config.set_delay_between_updates(0.2)
```
Comment (arhuaco) [inline src/js_test/test_ir_remote.js]: Since this driver doesn't provide updates you don't need set_delay_between_updates.

**Review by arhuaco — APPROVED** [review]
Please take a look at the comments.

File: src/driver_lirc.cpp (hunk 2 above)
Comment (arhuaco) [reply src/driver_lirc.cpp]: And this can be const as well.

File: src/driver_lirc.cpp (hunk 4 above)
Comment (eljuguetero) [reply src/driver_lirc.cpp]: done

File: src/driver_lirc.cpp (hunk 3 above)
Comment (eljuguetero) [reply src/driver_lirc.cpp]: done

**Pull request merged**